lib/tools/apks-utils.js

```diff
@@ -0,0 +1,213 @@
+ * @returns {string} the actual command stdout
+ * @throws {Error} If bundletool jar does not exist in PATH or there was an error while
+ * executing it
+ */
+apksUtilsMethods.execBundletool = async function (args, errorMsg) {
+  await this.initBundletool();
+  args = [
+    '-jar', this.binaries.bundletool,
+    ...args
+  ];
+  log.debug(`Executing bundletool with arguments: ${JSON.stringify(args)}`);
+  let stdout;
+  try {
+    ({stdout} = await exec(getJavaForOs(), args));
+    log.debug(`Command stdout: ${_.truncate(stdout, {length: 300})}`);
+    return stdout;
+  } catch (e) {
+    if (e.stdout) {
+      log.debug(`Command stdout: ${e.stdout}`);
+    }
+    if (e.stderr) {
+      log.debug(`Command stderr: ${e.stderr}`);
+    }
+    throw new Error(`${errorMsg}. Original error: ${e.message}`);
+  }
+};
+
+/**
+ * @param {string} specLocation - The full path to the generated device spec location
+ * @returns {string} The same `specLocation` value
+ * @throws {Error} If it is not possible to retrieve the spec for the current device
+ */
+apksUtilsMethods.getDeviceSpec = async function (specLocation) {
+  const args = [
+    'get-device-spec',
+    '--adb', this.executable.path,
+    '--device-id', this.curDeviceId,
+    '--output', specLocation,
+  ];
+  log.debug(`Getting the spec for the device '${this.curDeviceId}'`);
+  await this.execBundletool(args, 'Cannot retrieve the device spec');
+  return specLocation;
+};
+
+/**
+ * @typedef {Object} InstallApksOptions
+ * @property {?number|string} timeout [20000] - The number of milliseconds to wait until
+ *                                              the installation is completed
+ * @property {boolean} allowTestPackages [false] - Set to true in order to allow test
+ *                                                 packages installation.
+ * @property {boolean} useSdcard [false] - Set to true to install the app on sdcard
+ *                                         instead of the device memory.
+ * @property {boolean} grantPermissions [false] - Set to true in order to grant all the
+ *                                                permissions requested in the application's manifest
+ *                                                automatically after the installation is completed
+ *                                                under Android 6+.
+ */
+
+/**
+ * Installs the given .apks package into the device under test
+ *
+ * @param {string} apks - The full path to the .apks file
+ * @param {?InstallApksOptions} options - Installation options
+ * @throws {Error} If the .apks bundle cannot be installed
+ */
+apksUtilsMethods.installApks = async function (apks, options = {}) {
+  options = Object.assign({
+    timeout: APKS_INSTALL_TIMEOUT,
+  }, options, {replace: true});
+
+  const tmpRoot = await tempDir.openDir();
+  try {
+    const specPath = await this.getDeviceSpec(path.resolve(tmpRoot, 'deviceSpec.json'));
+    const args = [
+      'extract-apks',
```

Comment:
Couldn't call `--adb` here ✔️ 

```
Description:
    Extracts from an APK Set the APKs that should be installed on a given
    device.

Synopsis:
    bundletool extract-apks
        --apks=<archive.apks>
        --device-spec=<device-spec.json>
        --output-dir=<output-dir>
        [--instant]
        [--modules=<base,module1,module2>]

Flags:
    --apks: Path to the archive file generated by the 'build-apks' command.

    --device-spec: Path to the device spec file generated by the 'get-device-
        spec' command.

    --output-dir: Path to where the matched APKs will be extracted from the
        archive file.

    --instant: (Optional) When set, APKs of the instant modules will be
        extracted instead of the installable APKs.

    --modules: (Optional) List of modules to be extracted (defaults to all of
        them). Note that the dependent modules will also be extracted. Ignored
        if the device receives a standalone APK.
```

Reply:
thanks, will fix